File: src/exercises/chapter1/lesson6.rb
# frozen_string_literal: true

# S(n) = 1/1×2 + 1/2×3 +…+ 1/n x (n + 1)
# @param {Integer} n - The upper limit of the range (positive integer)
# @return {Float} - The sum of the series up to 1/n×(n + 1)

def solution(n)
  result = 0
  (1..n).each do |i|
    result += 1.0 / (i * (i + 1)) # Each element is 1 / (i * (i + 1))
  end
  result
end

File: src/exercises/chapter1/Lesson6.ts
/**
 * Lesson 6: Calculate S(n) = 1/(1×2) + 1/(2×3) + … + 1/(n×(n + 1))
 */
export default class Lesson6 {

  /**
   * Calculates the sum S(n) = 1/(1×2) + 1/(2×3) + … + 1/(n×(n + 1)).
   * @param {number} n - The upper limit number to calculate the sum for.
   * @return {number} - The sum of the series for the given n.
   */
  solution(n: number): number {
    // Initialize result to accumulate the sum
    let result = 0

    // Loop through all numbers from 1 to n and accumulate the sum of 1 / (i * (i + 1))
    for (let i = 1; i <= n; i++) {
      result += 1 / (i * (i + 1))
    }

    // Return the calculated sum
    return result
  }
}
